topostats/plotting.py

```diff
@@ -45,53 +87,51 @@ def __init__(
         alpha: float = 0.5,
         palette: str = "deep",
         savefig_format: str = "png",
-        output_dir: Union[str, Path] = ".",
+        output_dir: str | Path = ".",
         var_to_label: dict = None,
         hue: str = "basename",
     ) -> None:
-        """Initialise the class.
+        """
+        Initialise the class.
 
         Parameters
-        ==========
-        df: Union[pd.DataFrame]
+        ----------
+        df :  pd.DataFrame
             Pandas data frame of data to be summarised.
-        base_dir: Union[str, Path]
+        base_dir :  str | Path
             Base directory from which all paths are relative to.
-        csv_file: Union[str, Path]
+        csv_file :  str | Path
             CSV file of data to be summarised.
-        stat_to_sum: str
+        stat_to_sum :  str
             Variable to summarise.
-        molecule_id: str
+        molecule_id :  str
             Variable that uniquely identifies molecules.
-        image_id: str
+        image_id :  str
             Variable that uniquely identifies images.
-        hist: bool
+        hist :  bool
             Whether to plot histograms.
-        stat: str
+        stat :  str
             Statistic to plot on histogram 'count' (default), 'freq'.
-        bins: int
+        bins :  int
             Number of bins to plot.
-        kde: bool
+        kde :  bool
             Whether to include a Kernel Density Estimate.
-        cut: float = 20,
+        cut :  float = 20,
             Cut point for KDE.
-        figsize: tuple
+        figsize :  tuple
             Figure dimensions.
-        alpha: float
+        alpha :  float
             Opacity to use in plots.
-        palette: str = "deep"
+        palette :  str = "deep"
             Seaborn colour plot to use.
-        file_ext: str
+        savefig_format :  str
             File type to save plots as 'png' (default), 'pdf', 'svg'.
-        output_dir: Union[str, Path]
+        output_dir :  str | Path
             Location to save plots to.
-        var_to_label: dict
+        var_to_label :  dict
             Variable to label dictionary for automatically adding titles to plots.
-        hue: str
+        hue :  str
             Dataframe column to group plots by.
-
-        Returns
-        =======
         """
```

Comment:
No "Returns" heading provided here.

Reply:
That is what I've removed :wink:

Typically old is in red and new is green.

You might find it easier (I do) to not use the "Unified diff view" on pull requests where things are shown line by line and instead use the "Split diff view" which shows the side by side. There should be a little book symbol right above the very first file in this PR which toggles this, see also [Diff view options](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/about-comparing-branches-in-pull-requests#diff-view-options).

I also use the [Refined GitHub](https://github.com/refined-github/refined-github) browser plugin which seamlessly improves the GitHub User Experience. Many of their features have already been incorporated into GitHub (and are no longer part of the extension).